src/main/java/uk/gov/hmcts/reform/sscs/reference/data/model/Language.java

```diff
@@ -38,7 +39,15 @@ public boolean equals(Object o) {
             return false;
         }
         Language language = (Language) o;
-        return reference.equals(language.reference) && dialectReference.equals(language.dialectReference);
+
+        boolean referenceMatched = reference.equals(language.reference);
+        boolean dialectOrMrdReferenceMatched;
+        try {
+            dialectOrMrdReferenceMatched = dialectReference != null && dialectReference.equals(language.dialectReference);
+        } catch (NullPointerException e) {
+            dialectOrMrdReferenceMatched = mrdReference != null && mrdReference.equals(language.mrdReference);
+        }
+        return referenceMatched && dialectOrMrdReferenceMatched;

```

Comment:
Are we expecting the language.dialectReference to be null here? If we hit a null is it always an either or to compare against the mrdReference, what if language.mrdReference is null?